File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/lobby/LobbySettingsDto.java
package com.github.ryand6.sudokuweb.dto.entity.lobby;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.github.ryand6.sudokuweb.enums.Difficulty;
import com.github.ryand6.sudokuweb.enums.GameMode;
import com.github.ryand6.sudokuweb.enums.TimeLimitPreset;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class LobbySettingsDto {

    @JsonProperty("isPublic")
    private boolean isPublic;

    private Difficulty difficulty;

    private TimeLimitPreset timeLimit;

    private GameMode gameMode;

}

File: frontend/src/types/dto/entity/lobby/LobbySettingsDto.ts
import type { Difficulty } from "@/types/enum/Difficulty"
import type { GameMode } from "@/types/enum/GameMode"
import type { TimeLimitPreset } from "@/types/enum/TimeLimitPreset"

export type LobbySettingsDto = {
    isPublic: boolean,
    difficulty: Difficulty,
    timeLimit: TimeLimitPreset,
    gameMode: GameMode
}